src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,248 @@
+                    RunePosition_TestEquals(runePosition2, runePosition);
+                    RunePosition_TestEquals(runePosition3, runePosition);
+                    RunePosition_TestDeconstruct(runePosition);
+                }
+            }
+        }
+
+        [Fact]
+        public static void EnumerateRunePositions_Empty()
+        {
+            {
+                RunePosition.Utf16Enumerator enumerator = RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator();
+                Assert.Equal(default, enumerator.Current);
+                Assert.False(RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator().MoveNext());
+                Assert.Equal(default, enumerator.Current);
+            }
+            //Assert.False(RunePosition.EnumerateUtf8(ReadOnlySpan<char>.Empty).GetEnumerator().MoveNext());
+        }
+
+        [Theory]
+        [InlineData(new char[0], new int[0])] // empty
+        [InlineData(new char[] { 'x', 'y', 'z' }, new int[] { 'x', 'y', 'z' })]
+        [InlineData(new char[] { 'x', '\uD86D', '\uDF54', 'y' }, new int[] { 'x', 0x2B754, 'y' })] // valid surrogate pair
+        [InlineData(new char[] { 'x', '\uD86D', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // standalone high surrogate
+        [InlineData(new char[] { 'x', '\uDF54', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // standalone low surrogate
+        [InlineData(new char[] { 'x', '\uD86D' }, new int[] { 'x', 0xFFFD })] // standalone high surrogate at end of string
+        [InlineData(new char[] { 'x', '\uDF54' }, new int[] { 'x', 0xFFFD })] // standalone low surrogate at end of string
+        [InlineData(new char[] { 'x', '\uD86D', '\uD86D', 'y' }, new int[] { 'x', 0xFFFD, 0xFFFD, 'y' })] // two high surrogates should be two replacement chars
+        [InlineData(new char[] { 'x', '\uFFFD', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // literal U+FFFD
+        public static void EnumerateRunePositions_Battery16(char[] chars, int[] expected)
+        {
+            // Test data is smuggled as char[] instead of straight-up string since the test framework
+            // doesn't like invalid UTF-16 literals.
+
+            List<int> enumeratedValues = new List<int>();
+
+            RunePosition.Utf16Enumerator enumerator = RunePosition.EnumerateUtf16(chars).GetEnumerator();
+            Assert.Equal(default, enumerator.Current);
+
+            int expectedIndex = 0;
+            while (enumerator.MoveNext())
+            {
+                bool wasReplaced = Rune.DecodeFromUtf16(chars.AsSpan(expectedIndex), out Rune expectedRune, out int charsConsumed) != OperationStatus.Done;
+                RunePosition runePosition = enumerator.Current;
+
+                RunePosition_TestProps(expectedRune, expectedIndex, charsConsumed, wasReplaced, runePosition);
+
+                expectedIndex += charsConsumed;
+                enumeratedValues.Add(runePosition.Rune.Value);
+            }
+            Assert.Equal(default, enumerator.Current);
```

Comment:
Don't check the `Current` property since the enumeration is finished. Instead, check that the length you've been keeping track of this whole time properly covers the entire input span length. (Same comment in UTF8 test.)
 
```suggestion
            Assert.Equal(chars.Length, expectedIndex);
```